athenahealth/documents.go

```diff
@@ -176,3 +176,84 @@ func (h *HTTPClient) AddDocument(ctx context.Context, patientID string, opts *Ad
 
 	return res.DocumentID, nil
 }
+
+type AddPatientCaseDocumentOptions struct {
+	AutoClose          *bool
+	CallbackName       *string
+	CallbackNumber     *string
+	CallbackNumberType *string
+	DepartmentID       int
+	DocumentSource     string
+	DocumentSubclass   string
+	InternalNote       *string
+	OutboundOnly       *bool
+	Priority           *string
+	ProviderID         *int
+	Subject            *string
+}
+
+type addPatientCaseDocumentResponse struct {
+	PatientCaseID int `json:"patientcaseid"`
+}
+
+// AddDocument - Add patient case document for a patient
+// POST /v1/{practiceid}/patients/{patientid}/documents/patientcase
+// https://docs.athenahealth.com/api/api-ref/document-type-patient-case#Add-patient-case-document-for-a-patient
+func (h *HTTPClient) AddPatientCaseDocument(ctx context.Context, patientID string, opts *AddPatientCaseDocumentOptions) (int, error) {
+	var form url.Values
+
+	if opts != nil {
+		form = url.Values{}
+
+		if opts.AutoClose != nil && *opts.AutoClose {
+			form.Add("autoclose", "1")
+		}
+
+		if opts.CallbackName != nil {
+			form.Add("callbackname", *opts.CallbackName)
+		}
+
+		if opts.CallbackNumber != nil {
+			form.Add("callbacknumber", *opts.CallbackNumber)
+		}
+
+		if opts.CallbackNumberType != nil {
+			form.Add("callbacknumbertype", *opts.CallbackNumberType)
+		}
+
+		deptID := strconv.Itoa(opts.DepartmentID)
+		form.Add("departmentid", deptID)
```

Comment:
This technically isn't required since it's nested in this if block, should we always required `opts` to not be nil then optionally add non-required args?

Reply:
@willfitze It's a good suggestion, but I think this is the pattern we're following elsewhere in the repo, right?